IndexedDB/remove-listener-in-handler.any.js

```diff
@@ -0,0 +1,43 @@
+// META: script=support-promises.js
+
+const DB = databaseName("");
+
+const STORE = "store";
+
+// In the bug tickled by this test case, only two of the queries
+// worked and the rest stalled! Do five iterations just to make sure.
+const ITERATIONS = 5;
+
+const t = async_test("removeEventListener in a handler doesn't interfere with future transactions.");
+
+t.add_cleanup(() => {
+    indexedDB.deleteDatabase(DB);
+})
+
+let count = 0;
+
+const go = () => {
+    const dbReq = indexedDB.open(DB, 1);
+    dbReq.addEventListener("upgradeneeded", t.step_func((ev) => {
+        ev.target.result.createObjectStore(STORE, { autoIncrement: true });
```

Comment:
Is `autoIncrement` needed for the repro? If not, remove to minimize distraction. If so, comment about it?